src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/KeyValuePairListEncoding.cs

```diff
@@ -4,32 +4,39 @@ namespace Microsoft.VisualStudio.ProjectSystem.Debug;
 
 internal static class KeyValuePairListEncoding
 {
-    public static IEnumerable<(string Name, string Value)> Parse(string input)
+    /// <summary>
+    /// Parses the input string into a collection of key-value pairs with the given separator.
+    /// </summary>
+    /// <param name="input">The input string to parse.</param>
+    /// <param name="allowsEmptyKey">Indicates whether empty keys are allowed. If this is true, a pair will be returned if an empty key has a non-empty value. ie, =4</param>
```

Comment:
just curious, in what scenarios we allow empty keys?

Reply:
Really, when someone enters something like ;;;;CONSTANT;;;. It's just so we can gracefully handle that